cpp/src/sort/segmented_sort.cu

```diff
@@ -35,24 +34,49 @@ namespace {
  */
 enum class sort_method { STABLE, UNSTABLE };
 
-// returns segment indices for each element for all segments.
-// first segment begin index = 0, last segment end index = num_rows.
+/**
+ * @brief Builds indices to identify segments to sort
+ *
+ * The segments are added to the input table-view keys so they
+ * are lexicographically sorted within the segmented groups.
+ *
+ * ```
+ * Example 1:
+ * num_rows = 10
+ * offsets = {0, 3, 7, 10}
+ * segment-indices -> { 3,3,3, 7,7,7,7, 10,10,10 }
+ * ```
+ *
+ * ```
+ * Example 2: (offsets do not cover all indices)
+ * num_rows = 10
+ * offsets = {3, 7}
+ * segment-indices -> { 0,1,2, 7,7,7,7, 8,9,10 }
```

Comment:
I'm trying to get my head around the `8, 9, 10`. I realize that the actual label for these might be immaterial, so long as they are distinct from the other groups. But if these values appear as is in the output of `segmented_sorted_order()`, then would they not be incorrect as mentioned [here](https://github.com/rapidsai/cudf/pull/11888#discussion_r992626768)?

Reply:
The actual values do not mean much -- only their relative values are important. 
The `8,9,10` are just labels they just need to be distinct from the previous elements, sequential, and `>7`.
Hope that helps.

Reply:
> But if these values appear as is in the output of `segmented_sorted_order()`...

I should have checked the outputs of the test cases before making this comment. There is no problem here.

> Hope that helps.

Certainly. This looks good to me.